### DISCUSSION TITLE
Issue with TCP Data on Multi-UE

### DISCUSSION THREAD

**Discussion by houshangaziziaccelleran** [srsran/srsRAN_Project#608] [General Help]
Hi

Setup:

1. srsRAN Gnb latest release
2. OPEN5GS Core latest release
3. Amarisoft UE
4. B210 Radio

Test:

- Attaching 64 UEs and running TCP-DL on all UEs

Issues observed:

1.  `srsLog error - The backend queue size is about to reach its maximum capacity of 8192 elements, new log entries will get discarded.`
2.  Segmentation fault

**Note: I need to highlight that the same multi-UE test with the same log level is passed when doing UDP instead of TCP.**

Two sets of logs are attached.

Thanks
Houshang

[tcp-dl-segmentation-fault.zip](https://github.com/srsran/srsRAN_Project/files/15251510/tcp-dl-segmentation-fault.zip)
[tcp-dl-ues-lost.zip](https://github.com/srsran/srsRAN_Project/files/15251513/tcp-dl-ues-lost.zip)

**Comment by ismagom** [comment]
Hi @houshangaziziaccelleran , can you also attach the configuration you are using? 

Regarding the srsLog issue, that's not an issue, that's only saying that too many logs are being generated and are dropped. You can safely ignore it or reduce the log level or the layers logging. 

For the segmentation fault, could you run cmake with `-DCMAKE_BUILD_TYPE=RelWithDebInfo` and recompile, also make sure you have `backward-cpp` detected in cmake. This will help generate more meaningful backtrace when the segfault is detected. When you have that, could you paste here the output please. 

One more question, were you generating any channel impairment, it seems there are many UEs being disconnected.

**Comment by houshangaziziaccelleran** [comment]
Hi @ismagom 

1. Please find attached the config.
2. For the re-test, would `cmake -DCMAKE_BUILD_TYPE=RelWithDebInfo` followed by `make -j $(nproc) `help.
3. As for channel impairment question, I am just running TCP Data on UEs using iperf3. I am not using Amarisoft channel simulation: `channel_sim` is not set to `true`.

[config.txt](https://github.com/srsran/srsRAN_Project/files/15252022/config.txt)

**Reply by pgawlowicz** [reply]
Hi @houshangaziziaccelleran, any update on this issue? was it fixed with the newest project version?
